src/ActiveRoomsProvider.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { identity, sortBy } from 'lodash-es';
+import { addGlobalHook } from './common';
+import { PusherRoom } from './interfaces';
+
+const { withChatkit } = require('@pusher/chatkit-client-react');
+
+interface CreateRoomParams {
+  id?: string,
+  name: string,
+}
+
+export interface ActiveRoomsState {
+  loading: boolean,
+  activeRooms: PusherRoom[],
+  joinableRooms: PusherRoom[],
+  createRoom(room: CreateRoomParams): void,
+}
+
+export const ActiveRoomsContext = React.createContext<ActiveRoomsState>({
+  loading: true,
+  activeRooms: [],
+  joinableRooms: [],
+  createRoom: identity,
+});
+
+const sortRooms = (rooms: PusherRoom[]) => sortBy(rooms, ({ unreadCount, name }) => `${9999 - unreadCount}-${name}`);
+
+interface Props {
+  chatkit: any,
+  children: React.ReactChild,
+}
+
+export const ActiveRoomsProvider = withChatkit(({ chatkit, children }: Props) => {
+  const { currentUser, isLoading } = chatkit;
+  if (isLoading || !currentUser) {
+    return null;
+  }
+
+  const [loading, setLoading] = React.useState<boolean>(true);
+  const [activeRooms, setActiveRooms] = React.useState<PusherRoom[]>(sortRooms(currentUser.rooms));
+  const [joinableRooms, setJoinableRooms] = React.useState<PusherRoom[]>([]);
+  React.useEffect(() => {
+    chatkit.currentUser.enablePushNotifications();
+
+    const getRooms = async () => {
+      const joinableRooms = await currentUser.getJoinableRooms();
+      setJoinableRooms(joinableRooms);
+      setLoading(false);
+    };
+    getRooms();
+
+    const onAddedToRoom = (addedRoom: PusherRoom) => {
```

Comment:
You're creating these listeners but you're not unsubscribing from them - useEffect can return a cleanup function for unsubscribing and other cleanup tasks. This seems like a good use case for it, no?

Reply:
The lifecycle of `currentUser` is tied to the same as this (because we're wrapping ChatkitProvider)... But I will split this up and implement unhook on dismount